scene/gui/item_list.cpp

```diff
@@ -707,7 +723,7 @@ void ItemList::gui_input(const Ref<InputEvent> &p_event) {
 
 				if (diff < uint64_t(ProjectSettings::get_singleton()->get("gui/timers/incremental_search_max_interval_msec")) * 2) {
 					for (int i = current - 1; i >= 0; i--) {
-						if (items[i].text.begins_with(search_string)) {
+						if (items[i].text.begins_with(search_string) && CAN_SELECT(i)) {
```

Comment:
```suggestion
						if (CAN_SELECT(i) && items[i].text.begins_with(search_string)) {
```

I would swap it, since `CAN_SELECT(i)` is clearly faster than `begins_with`.

Reply:
Yeah that would be good to change before merging.